tests/headingcommand.js

```diff
@@ -171,13 +158,13 @@ describe( 'HeadingCommand', () => {
 				convertTo = option;
 			}
 
-			it( 'converts to default option when executed with already applied option', () => {
+			it( 'does nothing when executed with already applied option', () => {
 				const command = commands.heading1;
 
 				setData( document, '<heading1>foo[]bar</heading1>' );
 				command.execute();
 
-				expect( getData( document ) ).to.equal( '<paragraph>foo[]bar</paragraph>' );
+				expect( getData( document ) ).to.equal( '<heading1>foo[]bar</heading1>' );
```

Comment:
As it turned out, even without the condition `document` does not fire the `change` event. Probably it's smart enough to know this delta does nothing. To verify it, I used a custom batch passed to `execute()` with a spy on `rename` method. It's not a perfect assertion but I have no idea how to improve it.